source/extensions/filters/http/ip_tagging/ip_tagging_filter.cc

```diff
@@ -82,7 +82,13 @@ Http::FilterHeadersStatus IpTaggingFilter::decodeHeaders(Http::RequestHeaderMap&
 
   if (!tags.empty()) {
     const std::string tags_join = absl::StrJoin(tags, ",");
-    headers.appendEnvoyIpTags(tags_join, ",");
+
+    // Use the optional header instead of x-envoy-ip-tags if it's not empty.
+    if (!config_->ip_tag_header().get().empty()) {
```

Comment:
I think we need to sanitize this header so others can not spoof it. When you configure ip_tag_header as x-foo-tags, and the requests come in with x-foo-tags header, if the requests does not get any tag from envoy ip tagging filter, they still have x-foo-tags header and the value can be anything.

Reply:
at least we want to default to sanitize otherwise we   need to set security level to untrusted downstream. I didn't check metadata tho.

Reply:
@JuniorHsu @cqi1217 I think @nahratzah meant to post  https://github.com/envoyproxy/envoy/pull/36434#issuecomment-2408899011 here. let me know what you folks think about this.

Reply:
@alyssawilk we hit a dilemma here. 
currently ip_tagging is robust to untrusted downstream
https://github.com/envoyproxy/envoy/blob/a6848603acf28017f9194789b01d16c6c1782e3e/source/extensions/extensions_metadata.yaml#L438

however my understanding is non `x-` prefixed header won't be sanitized by envoy, which is vulnerable, so appending header could downgrade the security_posture.

However, @nahratzah mention the asymmetric behavior in https://github.com/envoyproxy/envoy/pull/36434#issuecomment-2408899011

What do you think?

Reply:
it'd help if I had clarity around why the user wants a custom header.  was it to bypass the x-envoy checks?  If it's just "we want a custom header" we could require an x-envoy prefix.  If they want a non-x-envoy header I agree this would be changing the security model and either that needs to be fixed, or (if intentioanl) needs a _lot_ more documentation